File: src/main/java/twao/Player.java
package twao;

import java.util.List;
import java.util.LinkedList;

public class Player {
    private final String                  nickname;
    private int                           numberOfVillages = 0;
    private int                           numberOfNobles;
    private final List<VillageAssignment> offAssignments = new LinkedList<>();
    private final List<VillageAssignment> fakeAssignments = new LinkedList<>();
    private final List<VillageAssignment> nobleAssignments = new LinkedList<>();
    private final List<VillageAssignment> fakeNobleAssignments = new LinkedList<>();

    /**
     * Initializes player.
     *
     * @param nickname          Player's nickname
     * @param numberOfNobles    Number of nobles that given player may produce
     */
    public Player(String nickname, int numberOfNobles) {
        this.nickname = nickname;
        this.numberOfNobles = numberOfNobles;
    }

    public String getNickname()             { return nickname; }

    public int getNumberOfVillages()        { return numberOfVillages; }
    public void increaseNumberOfVillages()  { numberOfVillages++; }
    public void decreaseNumberOfVillaes()   { numberOfVillages--; }

    public int getNumberOfNobles()          { return numberOfNobles; }
    public boolean hasNoble()               { return (numberOfNobles > 0) ? true : false; }

    /**
     * Decreases number of available nobles by 1.
     */
    public void delegateNoble()             { numberOfNobles--; }

    public List<VillageAssignment> getOffAssignmentsCopy()          { return new LinkedList<>(offAssignments); }
    public List<VillageAssignment> getFakeAssignmentsCopy()         { return new LinkedList<>(fakeAssignments); }
    public List<VillageAssignment> getNobleAssignmentsCopy()        { return new LinkedList<>(nobleAssignments); }
    public List<VillageAssignment> getFakeNobleAssignmentsCopy()    { return new LinkedList<>(fakeNobleAssignments); }

    public synchronized void putOffAssignment       (VillageAssignment assignment) { offAssignments.add(assignment); }
    public synchronized void putFakeAssignment      (VillageAssignment assignment) { fakeAssignments.add(assignment); }
    public synchronized void putNobleAssignment     (VillageAssignment assignment) { nobleAssignments.add(assignment); }
    public synchronized void putFakeNobleAssignment (VillageAssignment assignment) { fakeNobleAssignments.add(assignment); }
}

File: src/main/java/twao/Player.kt
package twao

data class Player(val nickname: String, private var _numberOfNobles: Int = 0) {
    private val offAssignments = mutableListOf<VillageAssignment>()
    private val fakeAssignments = mutableListOf<VillageAssignment>()
    private val nobleAssignments = mutableListOf<VillageAssignment>()
    private val fakeNobleAssignments = mutableListOf<VillageAssignment>()

    val offAssignmentsCopy
        get() = offAssignments.toList()
    val fakeAssignmentsCopy
        get() = fakeAssignments.toList()
    val nobleAssignmentsCopy
        get() = nobleAssignments.toList()
    val fakeNobleAssignmentsCopy
        get() = fakeNobleAssignments.toList()

    val numberOfNobles
        get() = _numberOfNobles
    var numberOfVillages: Int = 0
        private set

    fun increaseNumberOfVillages() {
        numberOfVillages++
    }

    fun decreaseNumberOfVillaes() {
        numberOfVillages--
    }

    fun hasNoble(): Boolean
        = numberOfNobles > 0

    fun delegateNoble() {
        _numberOfNobles--
    }

    @Synchronized fun putOffAssignment(assignment: VillageAssignment) {
        offAssignments.add(assignment)
    }

    @Synchronized fun putFakeAssignment(assignment: VillageAssignment) {
        fakeAssignments.add(assignment)
    }

    @Synchronized fun putNobleAssignment(assignment: VillageAssignment) {
        nobleAssignments.add(assignment)
    }

    @Synchronized fun putFakeNobleAssignment(assignment: VillageAssignment) {
        fakeNobleAssignments.add(assignment)
    }
}